src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogJenkinsPublisher.java

```diff
@@ -73,24 +76,48 @@ protected void doRun() throws Exception {
             tags = TagsUtil.addTagToTags(tags, "jenkins_url", DatadogUtilities.getJenkinsUrl());
 
             long projectCount = 0;
-            try {
-                projectCount = Jenkins.getInstance().getAllItems(Project.class).size();
-            } catch (NullPointerException e){
+            Jenkins instance = Jenkins.getInstanceOrNull();
+            if (instance == null) {
                 logger.fine("Could not retrieve projects");
+            } else {
+                projectCount = instance.getAllItems(Project.class).size();
             }
-            long pluginCount = 0;
-            try {
-                pluginCount = Jenkins.getInstance().pluginManager.getPlugins().size();
-            } catch (NullPointerException e){
-                logger.fine("Could not retrieve plugins");
-            }
-            client.gauge("jenkins.project.count", projectCount, hostname, tags);
-            client.gauge("jenkins.plugin.count", pluginCount, hostname, tags);
 
+            PluginData pluginData = collectPluginData();
+            client.gauge("jenkins.project.count", projectCount, hostname, tags);
+            client.gauge("jenkins.plugin.count", pluginData.getCount(), hostname, tags);
+            client.gauge("jenkins.plugin.active", pluginData.getActive(), hostname, tags);
+            client.gauge("jenkins.plugin.failed", pluginData.getFailed(), hostname, tags);
+            client.gauge("jenkins.plugin.inactivate", pluginData.getInactive(), hostname, tags);
+            client.gauge("jenkins.plugin.withUpdate", pluginData.getUpdatable(), hostname, tags);
         } catch (Exception e) {
             DatadogUtilities.severe(logger, e, null);
         }
-
     }
 
+    private PluginData collectPluginData() {
+        PluginData.Builder pluginData = PluginData.newBuilder();
+
+        Jenkins instance = Jenkins.getInstanceOrNull();
```

Comment:
Can you pass the jenkins instance as an argument of the collectPluginData method instead?